benchmark/Snakefile

```diff
@@ -0,0 +1,537 @@
+
+    def allele_counts(ts, sample_sets=None):
+        if sample_sets is None:
+            sample_sets = [ts.samples()]
+        def f(x):
+            return x
+        return ts.sample_count_stat(sample_sets, f, len(sample_sets),
+                                    span_normalise=False, windows='sites',
+                                    polarised=True, mode='site', strict=False)
+
+    mut_type = np.zeros(ts.num_sites)
+    for j, s in enumerate(ts.sites()):
+        mt = []
+        for m in s.mutations:
+            for md in m.metadata["mutation_list"]:
+                mt.append(md["mutation_type"])
+        if len(set(mt)) > 1:
+            mut_type[j] = 3
+        else:
+            mut_type[j] = mt[0]
+
+    freqs = allele_counts(ts, [sample_sets])
+    freqs = freqs.flatten().astype(int)
+    mut_afs = np.zeros((len(sample_sets)+1, 3), dtype='int64')
+    for k in range(3):
+        mut_afs[:, k] = np.bincount(freqs[mut_type == k+1], minlength=len(sample_sets) + 1)
+
+    return mut_afs
+
+rule generate_fs:
+    input:
+        output_dir + "/simulated_data/{seeds}/sim_OutOfAfrica_3G09_HuberDFE_{chrms}.trees"
+    output:
+        #output_dir + "/dadi/{seeds}/{chrms}/pop{ids}/pop{ids}.neu.fs",
+        #output_dir + "/dadi/{seeds}/{chrms}/pop{ids}/pop{ids}.nonneu.fs",
+        #output_dir + "/polyDFE/{seeds}/{chrms}/pop{ids}/pop{ids}.polyDFE.fs",
+        output_dir + "/dfe-alpha/{seeds}/{chrms}/pop{ids}/pop{ids}.dfe-alpha.neu.config",
+        output_dir + "/dfe-alpha/{seeds}/{chrms}/pop{ids}/pop{ids}.dfe-alpha.nonneu.config",
+        output_dir + "/dfe-alpha/{seeds}/{chrms}/pop{ids}/pop{ids}.dfe-alpha.fs"
+#        "results/anavar/pop0.anavar.ctl.txt",
+#        "results/prfreq/pop0.prfreq.fs"
+    threads: 1
+    run:
+        contig = species.get_contig(wildcards.chrms)
+        seq_len = contig.recombination_map.sequence_length
+
+        ts = tskit.load(input[0])
+        index = int(wildcards.ids)
+        sizes = np.cumsum(num_samples_per_population)
+        if index != 0: samps = ts.samples()[sizes[index-1]:sizes[index]]
+        else: samps = ts.samples()[:sizes[index]]
+        mut_afs = generate_fs_from_ts(ts, samps)
+
+        ## For dadi
+        nonneu_fs = dadi.Spectrum(mut_afs[:,0])
+        neu_fs = dadi.Spectrum(mut_afs[:,1])
+        neu_fs.to_file(output[0])
+        nonneu_fs.to_file(output[1])
+
+        ## For polyDFE
+        ## Project down to 20 chromosomes
+        ## See https://github.com/paula-tataru/polyDFE/issues/3
+        neu_fs_proj = neu_fs.project([20])
+        nonneu_fs_proj = nonneu_fs.project([20])
+        neu_len = round(seq_len * 0.3)
+        nonneu_len = round(seq_len * 0.7)
+        with open(output[2], 'w') as o:
+            o.write("1 1 20\n")
+            o.write(" ".join([str(round(f)) for f in neu_fs_proj[1:-1]]) + " " + str(neu_len) + "\n")
+            o.write(" ".join([str(round(f)) for f in nonneu_fs_proj[1:-1]]) + " " + str(nonneu_len) + "\n")
```

Comment:
Hi @chriscrsmith, here are the codes to generate frequency spectra for dadi and polyDFE, because they use different formats.